hunting/aws/docs/iam_unusual_access_key_usage_for_user.md

```diff
@@ -0,0 +1,63 @@
+# AWS IAM Unusual AWS Access Key Usage for User
+
+---
+
+## Metadata
+
+- **Author:** Elastic
+- **Description:** This hunting query gathers data from AWS CloudTrail logs to identify unusual AWS access key usage for a user. By detecting instances where an access key is used infrequently for a specific AWS event, this query helps identify potential misuse or abuse of AWS access keys. Adversaries may use access keys to gain unauthorized access to AWS resources, exfiltrate data, or perform other malicious activities within the environment.
+
+- **UUID:** `18ce3dbc-b1b3-11ef-9e63-f661ea17fbce`
+- **Integration:** [aws.cloudtrail](https://docs.elastic.co/integrations/aws/cloudtrail)
+- **Language:** `[ES|QL]`
+- **Source File:** [AWS IAM Unusual AWS Access Key Usage for User](../queries/iam_unusual_access_key_usage_for_user.toml)
+
+## Query
+
+```sql
+FROM logs-aws.cloudtrail*
+// Limit the search to the last 14 days
+| WHERE @timestamp > now() - 14 day
+| WHERE
+    // Filter for successful AWS CloudTrail events
+    event.dataset == "aws.cloudtrail"
+    and event.outcome == "success"
+
+    // Filter for AWS CloudTrail events with user identity and access key information
+    and aws.cloudtrail.user_identity.access_key_id IS NOT NULL
+    and aws.cloudtrail.resources.arn IS NOT NULL
+
+    // Ignore GetObject events
+    and event.action NOT IN ("GetObject")
+
+    // Filter out known service roles; expand this as needed
+    and NOT aws.cloudtrail.user_identity.arn LIKE "*AWSServiceRoleForConfig*"
+    and NOT aws.cloudtrail.user_identity.arn LIKE "*Elastic-Cloud-Security-Posture*"
+    and NOT aws.cloudtrail.user_identity.arn LIKE "*AmazonSSMRoleForInstancesQuickSetup*"
+
+| STATS
+    // Count the number of events for each daily bucket, user identity, access key, resource, and action
+    api_counts = count(*) by aws.cloudtrail.user_identity.arn, aws.cloudtrail.user_identity.access_key_id, event.action
+
+// Filter for access keys with less than 2 API calls per day
+| WHERE api_counts < 2
```

Comment:
this logic is already good, you can also look for instances where the same access key is used by two or more unique users (`count_distinct(arn) by access_key_id`) and where one of which has a high count of api calls  and the other one is low (someone stole the key and using it in parallel).

Reply:
Excellent idea Samir, I'll get that added. Thanks for the review!